lib/pbench/server/api/__init__.py

```diff
@@ -272,21 +273,66 @@ class Upload(Resource):
     def post(self):
         global app
         if not request.headers.get("filename"):
-            app.logger.debug("Missed filename in header")
-            abort(400, message="Missing filename header in request")
+            app.logger.debug(
+                "Tarfile upload: Post operation failed due to missing filename header"
+            )
+            abort(
+                400,
+                message="Missing filename header, POST operation requires a filename header to name the uploaded file",
+            )
         filename = secure_filename(request.headers.get("filename"))
 
         if not request.headers.get("md5sum"):
-            app.logger.debug("Missed md5sum in header")
-            abort(400, message="Missing md5sum header in request")
+            app.logger.debug(
+                f"Tarfile upload: Post operation failed due to missing md5sum header for file {filename}"
+            )
+            abort(
+                400,
+                message="Missing md5sum header, POST operation requires md5sum of an uploaded file in header",
+            )
         md5sum = request.headers.get("md5sum")
 
         app.logger.debug("Receiving file: {}", filename)
         if not allowed_file(filename):
-            app.logger.debug("Bad file extension received")
+            app.logger.debug(
+                f"Tarfile upload: Bad file extension received for file {filename}"
+            )
             abort(400, message="File extension not supported. Only .xz")
 
+        try:
+            content_length = int(request.headers.get("Content-Length"))
+        except ValueError:
+            app.logger.debug(
+                f"Tarfile upload: Invalid content-length header, not an integer for file {filename}"
+            )
+            abort(400, message="Invalid content-length header, not an integer")
+        except Exception:
+            app.logger.debug(
+                f"Tarfile upload: No Content-Length header value found for file {filename}"
+            )
+            abort(400, message="Missing required content-length header")
+        else:
+            max_upload_size = filesize_bytes(app.config["MAX_CONTENT_LENGTH"])
```

Comment:
Should we consider moving the conversion of the `MAX_CONTENT_LENGTH` value to a proper integer once when we load the server configuration?  Seems like a bad config file value would go unnoticed until the server started running and uploads started flowing in.

Reply:
Indeed, I think that would be a good idea, I will move the conversion in `create_app`